google/resource_compute_instance.go

```diff
@@ -495,6 +495,33 @@ func resourceComputeInstance() *schema.Resource {
 				},
 			},
 
+			"shielded_vm_config": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
```

Comment:
```suggestion
				Optional: true,
				// Since this block is used by the API based on which
				// image being used, the field needs to be marked as Computed.
				Computed: true
```

Reply:
Do you mind adding this, including the comment? While we're trying to avoid uses of `Optional` + `Computed` together in the provider going forward, this is an appropriate place to use it.

Because the API sets values by default when the block is unsent in the request, users will see a diff if they're using a shielded image without this block defined. We can't require them to define it though, because that would count as a breaking change. Computed here means that if the field is undefined in config, Terraform won't try and make a diff.

Reply:
given that, should this field also be added here? https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_instance_from_template.go#L47

Reply:
Nope- that's for a specific class of field where having the field defined as null is distinct from the field not being defined. That's not the case here.